<launch>
	<arg name="launch_ros_bridge"         default="true"/>
	<arg name="sigverse_ros_bridge_port"  default="50001" />
	<arg name="ros_bridge_port"           default="9090" />

	<node name="mob_avatar_controller" pkg="mob_avatar_controller" type="mob_avatar_controller" output="screen" launch-prefix="gnome-terminal -t 'Mob Avatar Controller' -- ">

		<!--
		<remap from="/mob/message/from_ros"      to="/aa" />
		<remap from="/mob/message/from_sigverse" to="/bb" />
		<remap from="/mob/avatar_transform"      to="/cc" />
		-->
	</node>

	<group if="$(arg launch_ros_bridge)">
		<group ns="sigverse_ros_bridge">
			<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
		</group>
		
		<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
			<arg name="port" value="$(arg ros_bridge_port)"/>
		</include>
	</group>
</launch>

